pysb/simulator/ssa_base.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import print_function
+
+import logging
+import re
+
+import numpy as np
+import sympy
+
+from pysb.bng import generate_equations
+from pysb.core import Expression
+from pysb.logging import setup_logger
+from pysb.simulator.base import Simulator, SimulatorException
+
+
+class SSABase(Simulator):
+    _supports = {'multi_initials': True, 'multi_param_values': True}
+
+    def __init__(self, model, verbose=False, tspan=None, **kwargs):
+        super(SSABase, self).__init__(model, verbose, **kwargs)
+
+        generate_equations(self._model)
+
+        self.tspan = tspan
+        self.verbose = verbose
+
+        # private attribute
+        self._parameter_number = len(self._model.parameters)
+        self._n_species = len(self._model.species)
+        self._n_reactions = len(self._model.reactions)
+        self._step_0 = True
+        self.num_sim = None
+        if verbose:
+            setup_logger(logging.INFO)
+
+    def _get_template_args(self):
```

Comment:
There's a lot of regex magic in this function. Please could you add a comment or two to describe what's going on?

Reply:
Added

Reply:
Added comments.